containers/daap-python-base/src/var/task/data_platform_paths.py

```diff
@@ -77,71 +109,130 @@ def get_account_id() -> str:
     return boto3.client("sts").get_caller_identity()["Account"]
 
 
-class DataProductConfig:
+@dataclass
+class DataProductElement:
     """
-    Configures the name, S3 paths, and athena table names for a data product.
+    A dataset within the data product. The curated data for each element
+    is queryable in its own athena table.
     """
 
-    def __init__(self, name: str, table_name: str, bucket_name: str | None = None):
-        """
-        Generate all the paths based on the data product name and a table name
-        """
-        if bucket_name is None:
-            bucket_name = get_bucket_name()
+    name: str
+    data_product: DataProductConfig
 
-        self.name = name
+    @staticmethod
+    def load(name, data_product_name):
+        data_product = DataProductConfig(name=data_product_name)
+        return DataProductElement(data_product=data_product, name=name)
```

Comment:
```suggestion
    def load(element_name, data_product_name):
        data_product = DataProductConfig(name=data_product_name)
        return DataProductElement(data_product=data_product, name=element_name)
```
Specifying `element_name` as this is a static method, so it's clear outside the context of the element class.